client.go

```diff
@@ -227,7 +227,9 @@ func NewClient(options ...ClientOptionFunc) (*Client, error) {
 
 	if c.snifferEnabled {
 		// Sniff the cluster initially
-		if err := c.sniff(c.snifferTimeoutStartup); err != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		if err := c.sniff(ctx, c.snifferTimeoutStartup); err != nil {
```

Comment:
Please remove this and keep the code near to what we have in elastic.v5.

This opens more questions, but does not solve any issues. E.g. why a timeout of 10 seconds? Why cancel the context via timeout _and_ pass a timeout?